Mathlib/Data/BitVec/Lemmas.lean

```diff
@@ -218,4 +231,58 @@ instance : CommSemiring (BitVec w) :=
     toFin_zero toFin_one toFin_add toFin_mul (Function.swap toFin_nsmul)
     toFin_pow toFin_natCast
 
+/-!
+## Extensionality
+-/
+
+/-- If two bitvectors agree on all in-bound bits, then they agree on all bits -/
+theorem getLsb_eq_of_getLsb' {x y : BitVec w} (h : ∀ (i : Fin w), x.getLsb' i = y.getLsb' i) :
+    ∀ (i : ℕ), x.getLsb i = y.getLsb i := by
+  simp only [getLsb, testBit]
+  intro i
+  by_cases hi : i < w
```

Comment:
I think `cases lt_or_le i w` gives a nicer proof.

Reply:
I changed the proof to use `lt_or_le`